frontends/web/src/routes/account/send/send.tsx

```diff
@@ -774,25 +772,12 @@ class Send extends Component<Props, State> {
               />
               <MessageWaitDialog isShown={isSent} messageType="sent" />
               <MessageWaitDialog isShown={isAborted} messageType="abort" />
-              <Dialog
-                open={activeScanQR}
-                title={t('send.scanQR')}
-                onClose={this.toggleScanQR}>
-                {videoLoading && <Spinner guideExists />}
-                <video
-                  id="video"
-                  width={400}
-                  height={300 /* fix height to avoid ugly resize effect after open */}
-                  className={style.qrVideo}
-                  onLoadedData={this.handleVideoLoad} />
-                <div className={['buttons', 'flex', 'flex-row', 'flex-between'].join(' ')}>
-                  <Button
-                    secondary
-                    onClick={this.toggleScanQR}>
-                    {t('button.back')}
-                  </Button>
-                </div>
-              </Dialog>
+              <DialogScanQR
+                activeScanQR={activeScanQR}
+                onLoadedVideo={this.handleVideoLoad}
+                toggleScanQR={this.toggleScanQR}
+                videoLoading={videoLoading}
+              />
```

Comment:
I did a quick test, looks like the videoLoading state could be moved into `DialogScanQR` component.

```diff
diff --git a/frontends/web/src/routes/account/send/components/dialogs/scan-qr-dialog.tsx b/frontends/web/src/routes/account/send/components/dialogs/scan-qr-dialog.tsx
index 4515d079f..d7fb50dcc 100644
--- a/frontends/web/src/routes/account/send/components/dialogs/scan-qr-dialog.tsx
+++ b/frontends/web/src/routes/account/send/components/dialogs/scan-qr-dialog.tsx
@@ -1,3 +1,4 @@
+import { useEffect, useState } from 'react';
 import { useTranslation } from 'react-i18next';
 import { Dialog } from '../../../../../components/dialog/dialog';
 import { Button } from '../../../../../components/forms';
@@ -7,24 +8,33 @@ import style from '../../send.module.css';
 type TProps = {
     activeScanQR: boolean;
     toggleScanQR: () => void;
-    videoLoading: boolean;
-    onLoadedVideo: () => void;
 }
 
-const DialogScanQR = ({ activeScanQR, toggleScanQR, videoLoading, onLoadedVideo }: TProps) => {
+const DialogScanQR = ({
+  activeScanQR,
+  toggleScanQR,
+}: TProps) => {
   const { t } = useTranslation();
+  const [videoLoading, setVideoLoading] = useState(true);
+  useEffect(() => {
+    if (activeScanQR) {
+      setVideoLoading(true);
+    }
+  }, [activeScanQR]);
   return (
     <Dialog
       open={activeScanQR}
       title={t('send.scanQR')}
       onClose={toggleScanQR}>
       {videoLoading && <Spinner guideExists />}
-      <video
-        id="video"
-        width={400}
-        height={300 /* fix height to avoid ugly resize effect after open */}
-        className={style.qrVideo}
-        onLoadedData={onLoadedVideo} />
+      {activeScanQR && (
+        <video
+          id="video"
+          width={400}
+          height={300 /* fix height to avoid ugly resize effect after open */}
+          className={style.qrVideo}
+          onLoadedData={() => setVideoLoading(false)} />
+      )}
       <div className={['buttons', 'flex', 'flex-row', 'flex-between'].join(' ')}>
         <Button
           secondary
diff --git a/frontends/web/src/routes/account/send/send.tsx b/frontends/web/src/routes/account/send/send.tsx
index 693a5da0c..574386aad 100644
--- a/frontends/web/src/routes/account/send/send.tsx
+++ b/frontends/web/src/routes/account/send/send.tsx
@@ -92,7 +92,6 @@ interface State {
     activeCoinControl: boolean;
     hasCamera: boolean;
     activeScanQR: boolean;
-    videoLoading: boolean;
     note: string;
 }
 
@@ -125,7 +124,6 @@ class Send extends Component<Props, State> {
     activeCoinControl: false,
     hasCamera: false,
     activeScanQR: false,
-    videoLoading: false,
     note: '',
     customFee: '',
   };
@@ -540,7 +538,6 @@ class Send extends Component<Props, State> {
     }
     this.setState({
       activeScanQR: true,
-      videoLoading: true,
     }, () => {
       this.qrCodeReader && this.qrCodeReader
         .decodeFromInputVideoDevice(undefined, 'video')
@@ -564,10 +561,6 @@ class Send extends Component<Props, State> {
     this.setState({ activeCoinControl: false });
   };
 
-  private handleVideoLoad = () => {
-    this.setState({ videoLoading: false });
-  };
-
   public render() {
     const { t, code } = this.props;
     const {
@@ -599,7 +592,6 @@ class Send extends Component<Props, State> {
       activeCoinControl,
       hasCamera,
       activeScanQR,
-      videoLoading,
       note,
     } = this.state;
 
@@ -774,9 +766,7 @@ class Send extends Component<Props, State> {
               <MessageWaitDialog isShown={isAborted} messageType="abort" />
               <DialogScanQR
                 activeScanQR={activeScanQR}
-                onLoadedVideo={this.handleVideoLoad}
                 toggleScanQR={this.toggleScanQR}
-                videoLoading={videoLoading}
               />
             </View>
           </Main>
```

Reply:
This made me thinking how the result of the scanning video gets into the main component, as there ins no onResult callback...

The element is found by `this.qrCodeReader.decodeFromInputVideoDevice(undefined, 'video')` (deprecated). 

Maybe also out of scope for this PR but it feels that more code related scanning i.e. `this.qrCodeReader` could be moved to `DialogScanQR`.

Reply:
> Maybe also out of scope for this PR but it feels that more code related scanning i.e. this.qrCodeReader could be moved to DialogScanQR.

for this one, I'll try to address it in a different PR.